src/mono_font/mapping.rs

```diff
@@ -0,0 +1,377 @@
+pub struct StrGlyphMapping<'a> {
+    data: &'a str,
+    replacement_index: usize,
+}
+
+impl<'a> StrGlyphMapping<'a> {
+    /// Creates a new glyph mapping.
+    pub const fn new(data: &'a str, replacement_index: usize) -> Self {
+        Self {
+            data,
+            replacement_index,
+        }
+    }
+
+    /// Returns an iterator over the character ranges.
+    pub fn ranges(&self) -> impl Iterator<Item = (usize, RangeInclusive<char>)> + '_ {
+        let mut chars = self.data.chars();
+        let mut index = 0;
+
+        core::iter::from_fn(move || {
+            let start_index = index;
+
+            let range = match chars.next()? {
+                '\0' => {
+                    let start = chars.next()?;
+                    let end = chars.next()?;
+
+                    index += end as usize - start as usize + 1;
+
+                    start..=end
+                }
+                c => {
+                    index += 1;
+
+                    c..=c
+                }
+            };
+
+            Some((start_index, range))
+        })
+    }
+
+    /// Returns an enumerated iterator over the characters.
+    pub fn chars(&self) -> impl Iterator<Item = (usize, char)> + '_ {
+        let mut chars = self.data.chars();
+
+        core::iter::from_fn(move || {
+            let range = match chars.next()? {
+                '\0' => {
+                    let start = chars.next()?;
+                    let end = chars.next()?;
+
+                    start as u32..=end as u32
+                }
+                c => c as u32..=c as u32,
+            };
+
+            Some(range)
+        })
+        .flatten()
+        //MSRV: directly iterator over char ranges
+        .filter_map(core::char::from_u32)
+        .enumerate()
+    }
+
+    /// Returns if the mapping contains the given char.
+    pub fn contains(&self, c: char) -> bool {
+        self.chars().any(|(_, v)| c == v)
+    }
+}
+
+impl GlyphMapping for StrGlyphMapping<'_> {
+    fn index(&self, c: char) -> usize {
+        // PERF: use ranges instead of chars iter
+        self.chars()
+            .find(|(_, v)| c == *v)
+            .map(|(index, _)| index)
+            .unwrap_or(self.replacement_index)
+    }
+}
+
+macro_rules! impl_mapping {
+    ($( $(#[$meta:meta])* ($enum_variant:ident, $constant:ident, $mapping:expr), )*) => {
+        /// Mapping.
+        ///
+        /// This enum lists all mappings that are included in embedded-graphics. It is used
+        /// to automatically generate font data for all mappings and isn't normally used in
+        /// applications.
+        #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+        pub enum Mapping {
+            $(
+                $(#[$meta])*
+                $enum_variant,
+            )*
+        }
+
+        impl Mapping {
+            /// Returns an iterator over all mappings.
+            pub fn all() -> impl Iterator<Item = Self> {
```

Comment:
I'd prefer `into_iter()` here.

Reply:
Sorry, I had missed this comment. Why do you think that `into_iter` would be better? For me `into_iter` should be used to convert a value into an iterator, but this function doesn't depend on a value.

Reply:
@rfuest I guess github doesn't like to notify about comments (as I missed your response as well).

Technically, Mapping _is_ a "value" (well, an enum, sure) and this method returns an iterator to this. Maybe I should change my preference to `iter()` as there is no actual conversion.

My main issue with `all()` is that `all` is an iterator method and if you ever have to write `Mapping::all().all([...])` it gets at least visually confusing.

Reply:
> My main issue with `all()` is that `all` is an iterator method and if you ever have to write `Mapping::all().all([...])` it gets at least visually confusing.

That does look confusing, but IMO it's very unlikely that anyone will ever need to use this construct.

I'm be OK with changing it to `iter`. While all `iter` methods in the standard library are methods and not associated functions, the `strum` crate does also use `iter` in the same way.

If the name bothers you enough feel free to open a PR.